scalalib/src/mill/javalib/android/AndroidLibModule.scala

```diff
@@ -0,0 +1,113 @@
+package mill.javalib.android
+
+import mill.*
+import mill.scalalib.*
+import mill.api.PathRef
+import mill.define.Task
+import mill.scalalib.publish.{PackagingType, PublishInfo}
+import os.RelPath
+import upickle.default.*
+
+@mill.api.experimental
+trait AndroidLibModule extends AndroidModule with PublishModule {
+
+  private val parent: AndroidLibModule = this
+
+  //  /**
+  //   * The packaging type. See [[PackagingType]] for specially handled values.
+  //   */
+  override def pomPackagingType: String =
+    this match {
+      case _: BomModule => PackagingType.Pom
+      case _ => PackagingType.Aar
+    }
+
+  override def jar: T[PathRef] = {
+    androidAar()
+  }
+
+  override def defaultPublishInfos: T[Seq[PublishInfo]] = {
+    def defaultPublishJars: Task[Seq[(PathRef, PathRef => PublishInfo)]] = {
+      pomPackagingType match {
+        case PackagingType.Pom => Task.Anon(Seq())
+        case _ => Task.Anon(Seq(
+            (jar(), PublishInfo.aar _),
+            (sourceJar(), PublishInfo.sourcesJar _),
+            (docJar(), PublishInfo.docJar _)
+          ))
+      }
+    }
+    Task {
+      defaultPublishJars().map { case (jar, info) => info(jar) }
+    }
+  }
+
+  def androidAar: T[PathRef] = Task {
+    val dest = T.dest
+    val aarFile = dest / "library.aar"
+    val compiledRes = dest / "compiled-res"
+    val classesJar = dest / "classes.jar"
+    val unpackedAar = dest / "unpacked-aar"
+
+    val classFiles = compile().classes.path
+    os.proc("jar", "cvf", classesJar.toString, "-C", classFiles.toString, ".").call()
```

Comment:
this can also be changed to use the `Jvm.createJar` util or maybe it will work with `super.jar()` task